File: flut/.flutter/lib/flutter/services/clipboard.dart
import 'package:flutter/services.dart';
import 'package:flut/flut/object.dart';
import 'package:flut/flut/runtime.dart';

class FlutClipboardData extends FlutValueObject {
  final ClipboardData data;
  const FlutClipboardData(this.data) : super('ClipboardData');

  @override
  Map<String, dynamic> flutEncode() {
    final result = flutBaseProps();
    result['text'] = data.text;
    return result;
  }

  static ClipboardData flutDecode(
    FlutRuntime runtime,
    Map<String, dynamic> data,
  ) {
    return ClipboardData(
      text: runtime.unpackRequiredField<String>(data, 'text'),
    );
  }
}

class FlutClipboard {
  FlutClipboard._();

  static void registerStatics(FlutRuntime runtime) {
    runtime.registerStatic('Clipboard.setData', setData);
    runtime.registerStatic('Clipboard.getData', getData);
  }

  static dynamic setData(FlutRuntime runtime, ClipboardData data) {
    Clipboard.setData(data);
  }

  static dynamic getData(FlutRuntime runtime, String format) {
    return Clipboard.getData(format);
  }
}

File: flut/flutter/services/clipboard.py
from typing import override

from flut._flut.engine import FlutValueObject, call_dart_static, _flut_pack_value


class ClipboardData(FlutValueObject):
    _flut_type = "ClipboardData"

    def __init__(self, *, text):
        super().__init__()
        self.text = text

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        result["text"] = _flut_pack_value(self.text)
        return result

    @staticmethod
    def _flut_unpack(data: dict) -> "ClipboardData":
        return ClipboardData(
            text=data["text"],
        )


class Clipboard:
    @staticmethod
    def setData(data: ClipboardData):
        call_dart_static("Clipboard", "setData", data._flut_pack())

    @staticmethod
    def getData(format: str):
        result = call_dart_static("Clipboard", "getData", format)
        if result is not None:
            return ClipboardData._flut_unpack(result)
        return None
